### PR TITLE
testcases/le_tc/kernel: Add protected build dependency in testcases to resolve asserts

### PR DESCRIPTION
**Pull request by NilabhShant**
libc_mqueue & Timer make direct calls to kernel specific structures
which results in assert.

So this patch adds config dependency for above mentioned kernel testcases.

Signed-off-by: Nilabh <n.shant@partner.samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e73042094c6fe42f5e561b6552027343842cfbfe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e73042094c6fe42f5e561b6552027343842cfbfe] - Code Rule Check Result:
apps/examples/testcase/le_tc/kernel/tc_timer.c:122: ERROR: [IDT_M_TAB] please, no space before tabs

File: apps/examples/testcase/le_tc/kernel/kernel_tc_main.c [hunk 1]
```diff
@@ -103,7 +103,7 @@ int tc_kernel_main(int argc, char *argv[])
 	libc_misc_main();
 #endif
 
-#ifdef CONFIG_TC_KERNEL_LIBC_MQUEUE
+#if !defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_TC_KERNEL_LIBC_MQUEUE)
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: I don't know why this is needed. The libc_mqueue_main uses syscall APIs or libc APIs. Could you let me know why this is needed?

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: And if this is needed indeed, let's add a dependency in Kconfig, CONFIG_TC_KERNEL_LIBC_MQUEUE. Not here.

File: apps/examples/testcase/le_tc/kernel/tc_timer.c [hunk 2]
```diff
@@ -273,7 +275,10 @@ static void tc_timer_timer_initialize(void)
 
 int timer_main(void)
 {
+#ifndef CONFIG_BUILD_PROTECTED
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/kernel/tc_timer.c]: Is CONFIG_BUILD_PROTECTED right? not APP_SEPARATION?

File: apps/examples/testcase/le_tc/kernel/kernel_tc_main.c (hunk 1 above)
Comment (NilabhShant) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: As libc_mqueue_main has 2 testecases 

1. tc_libc_mqueue_mq_getattr();
2. tc_libc_mqueue_mq_setattr();

and both are giving assert so I directly put main in config check.

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: mq_getattr and setattr are libc APIs. It should not be a problem. Important thing is not a blocking tc, we need to check whether it has a problem or not.

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: Could you check root cause of assert?

Comment (NilabhShant) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: These testcases internally accessing kernel specific structure to validate which is not allowed in user space in protected build configuration, that's why they are  giving assert. 

So I put this testcase under config check.

File: apps/examples/testcase/le_tc/kernel/tc_timer.c (hunk 2 above)
Comment (NilabhShant) [reply apps/examples/testcase/le_tc/kernel/tc_timer.c]: CONFIG_BUILD_PROTECTED is fine.

**Comment by seinfra** [comment]
Target : [06b19319d44aca72d0b3af8e6e6f90b4dcd94618] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [06b19319d44aca72d0b3af8e6e6f90b4dcd94618] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1db6673c096706a29c6a80277386c68a516a17ea] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/kernel_tc_main.c (hunk 1 above)
Comment (NilabhShant) [reply apps/examples/testcase/le_tc/kernel/kernel_tc_main.c]: As suggested I added dependency in Kconfig, Please check

**Comment by seinfra** [comment]
Target : [1db6673c096706a29c6a80277386c68a516a17ea] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [709a1c44fd73effd1fe5b8e4be172b51ec75e9f1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [709a1c44fd73effd1fe5b8e4be172b51ec75e9f1] - Code Rule Check (C++) OK.

**Comment by NilabhShant** [comment]
My mistake close the PR , Re-oping it

**Pull request closed without merging**